Skip RKMs for seitan key

mlsteele opened the pull request:
Tests coming soon

maxtaco commented on go/teams/teams.go:
```diff
@@ -282,6 +286,16 @@ func (t *Team) ApplicationKey(ctx context.Context, application keybase1.TeamAppl
+func (t *Team) UseRKMForApp(application keybase1.TeamApplication) bool {
+	switch application {
+	case keybase1.TeamApplication_SEITAN_INVITE_TOKEN:
+		// Seitan tokens do not use RKMs because implicit admins have all the priveleges of explicit members.
```
Misspelled "privileges"

mlsteele commented:
Changed it so that if AnnotateInvite fails it is logged and thrown out. This will trigger when any of us processes a team with seitan invites committed to a sigchain prior to this PR. Because they used different keys.

Added regression testing.
